Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -227,8 +228,16 @@ public IEnumerable<UnspentOutputReference> GetAllSpendableTransactions(CoinType
     /// </summary>
     public class AccountRoot
     {
-        public AccountRoot()
+        /// <summary>Provider of date time functionality.</summary>
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the wallet.
+        /// </summary>
+        /// <param name="dateTimeProvider">Provider of date time functionality.</param>
+        public AccountRoot(IDateTimeProvider dateTimeProvider)
```

Comment:
In this case I would prefer to not have `IDateTimeProvider` inside `AccountRoot`.
Rather pass the creation date to the constructor.

Reply:
yes i agree, i've made such decision elsewhere, but not here, but i don't like it either :)